## 50% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 3
Expected to fail tests: 1
Passed tests: 4
```
```
IDs of failed tests: 702745,727229,703133 
```
```
IDs of expected to fail tests: 702843 
```
<h3>Failed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=approval-fees-not-shown&PR_ID=21532&APK_NAME=21532.apk&TR_CASE_IDS=702745,727229,703133">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724352">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `show-profiles` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:245: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:192: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0b4eab4dc3384865b890df5975401779?auth=ef6605a01d7ad975a1273e06b9bd7d0c#86">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724350">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking //*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"] </div></blockquote></p>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:430: in handle_contact_request
    self.close_activity_centre.wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:145: in wait_for_visibility_of_element
    .until(expected_conditions.visibility_of_element_located((self.by, self.locator)))
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:86: in until
    value = method(self._driver)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:152: in _predicate
    return _element_if_visible(driver.find_element(*locator))
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:175: in _element_if_visible
    return element if element.is_displayed() == visibility else False
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webelement.py:81: in is_displayed
    return self._execute(Command.IS_ELEMENT_DISPLAYED)['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute
    return self._parent.execute(command, params)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 The test session has already finished, and can't receive further commands. For help please check https://docs.saucelabs.com/dev/error-messages
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/95a1cb33b920408a9524621b0e62b803?auth=e7508e944cc6455fe8a63725e8f752a2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/94e792c01b4c42298cf4dd7330109dd6?auth=89eb152c725e0b4f55add1fbd191903d">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724353">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='container']/android.widget.TextView[@text='Ether']/../android.widget.TextView[3]` </div><div>Device 2: `Text` is `0.11359 ETH` </div></blockquote></p>

```
critical/test_wallet.py:158: in test_wallet_send_eth
    self.errors.verify_no_errors()
base_test_case.py:192: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.3668 but expected to be 0.3669
```

<br/><br/></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724351">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Skipped due to waku issue on staging fleet]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (4)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724349">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0b4eab4dc3384865b890df5975401779?auth=ef6605a01d7ad975a1273e06b9bd7d0c#54">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724354">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724355">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3724356">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/swap/setup_swap/view.cljs
+++ b/src/status_im/contexts/wallet/swap/setup_swap/view.cljs
@@ -71,8 +71,9 @@
         max-slippage           (rf/sub [:wallet/swap-max-slippage])
         loading-swap-proposal? (rf/sub [:wallet/swap-loading-swap-proposal?])
         currency-symbol        (rf/sub [:profile/currency-symbol])
-        approval-required      (rf/sub [:wallet/swap-proposal-approval-required])
-        max-fee                (if approval-required
+        approval-required?     (rf/sub [:wallet/swap-proposal-approval-required])
+        approval-status        (rf/sub [:wallet/swap-approval-transaction-status])
+        max-fee                (if (and approval-required? (not= approval-status :confirmed))
                                  (money/add approval-gas-fee swap-proposal-fee)
                                  swap-proposal-fee)
         max-fee-formatted      (utils/fiat-formatted-for-ui

--- a/src/status_im/subs/wallet/swap_test.cljs
+++ b/src/status_im/subs/wallet/swap_test.cljs
@@ -257,7 +257,7 @@
       swap-data)
     (is (match? "0x10000" (rf/sub [sub-name])))))
 
-(h/deftest-sub :wallet/wallet-swap-proposal-fee-fiat-formatted
+(h/deftest-sub :wallet/wallet-swap-proposal-fee-fiat
   [sub-name]
   (testing "wallet send fee calculated and formatted in fiat"
     (swap! rf-db/app-db
@@ -266,9 +266,8 @@
            (assoc-in [:wallet :current-viewing-account-address] "0x1")
            (assoc-in [:wallet :ui :swap] swap-data)
            (assoc-in [:currencies] currencies)
-           (assoc-in [:profile/profile :currency] :usd)
-           (assoc-in [:profile/profile :currency-symbol] "$")))
+           (assoc-in [:profile/profile :currency] :usd)))
 
     (let [token-symbol-for-fees "ETH"
           result                (rf/sub [sub-name token-symbol-for-fees])]
-      (is (match? result "$1.00")))))
+      (is (match? result 1)))))

